src/theory/strings/theory_strings.cpp

```diff
@@ -198,6 +198,34 @@ void TheoryStrings::presolve() {
 // MODEL GENERATION
 /////////////////////////////////////////////////////////////////////////////
 
+/**
+ * Helper that allows ordering sequence types by how nested they are.
```

Comment:
I'd prefer not adding this utility.

Can't we just have a simple policy in the code along the lines of:  to compute model values for a type T, first check if it is (Seq T1), if so, compute model values for T1 first?